### ISSUE TITLE
Add possibility to extend editors

### ISSUE THREAD

**Issue by fechy** [saurabhdaware/projectman#4]
Currently, only popular editors are available.

Will be useful to allow custom editor commands.
For instance, I use `vim` for some projects, and `webstorm` for Web projects and `Idea` for Java, etc.

**Comment by saurabhdaware** [comment]
Yeah, we can add vim and webstorm since they are pretty popular too.

I give the option to `pm edit` which opens settings.json where you can set your text editor value to the "commandToOpen" key so you can basically use whichever text editor that you want to use so I'll leave the `Idea` to manual settings (Since there are way too many text editors and I don't really want to flood users with too many options). But yes let's add vim and webstorm!

Are you interested in working on this? I don't really have vim and webstorm installed right now. You'll just have to add following values in `lib/action.js#setEditor()` 
```json
{
    "name": "WebStorm",
    "value": "command that webstorm use to open directories (I guess its 'wstorm' but I never tried though)"
},
{
     "name": "Vim",
     "path": "vim"
}
```

**Comment by saurabhdaware** [comment]
Resolved this in 4d3237da745dd61c7d3faf8de211d8c3bad553b8 🎉
- Added `vim` in set editor
- Showing message in `pm seteditor` explaining how to add editors that are not listed.

Thanks for your issue @fechy :D

**Comment by saurabhdaware** [comment]
Will keep it open for #6

**Comment by nhammond101** [comment]
Instead of defining the editor in the json file, it would be very flexible and extensible to add a `.projectmanrc` in the project itself.

That way, each project would define it's own editor and potentially different versions of the same editor if required

**Comment by saurabhdaware** [comment]
It will get difficult to handle it from the codebase. I would still need `settings.json` to link to the project then I would have to open `.projectmanrc` inside that path which is inside the settings to read the configurations so for now I'll actually prefer
```js
const settingsData = {
    "commandToOpen": "code",
    "projects": [
        {
            "name":"MyCoolProject",
            "path":"/home/path/projects/mycoolproject",
            "editor":"vim"
        },
        {
            "name":"TwoProject",
            "path":"/path/something/project"
        }
    ]
}
```
This will also allow me to not make any major changes in the current documentation (since `pm edit` command is already there)

Do let me know what you think :D

**Comment by saurabhdaware** [comment]
Resolved in fe481498a8ffaf70e967dc4cac8bf14e56a33a55, Thank You @fechy for your input! Would love to work with you in future :)

Closing this.

**Comment by vzool** [comment]
Hi @saurabhdaware, is this update available now? Thanks

> It will get difficult to handle it from the codebase. I would still need `settings.json` to link to the project then I would have to open `.projectmanrc` inside that path which is inside the settings to read the configurations so for now I'll actually prefer
> 
> ```js
> const settingsData = {
>     "commandToOpen": "code",
>     "projects": [
>         {
>             "name":"MyCoolProject",
>             "path":"/home/path/projects/mycoolproject",
>             "editor":"vim"
>         },
>         {
>             "name":"TwoProject",
>             "path":"/path/something/project"
>         }
>     ]
> }
> ```
> 
> This will also allow me to not make any major changes in the current documentation (since `pm edit` command is already there)
> 
> Do let me know what you think :D

**Comment by saurabhdaware** [comment]
Yes @vzool It is available, I just published it yesterday under v1.1.0 :)
